core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -486,6 +489,10 @@ public int propertyAsInt(String property, int defaultValue) {
     return PropertyUtil.propertyAsInt(properties, property, defaultValue);
   }
 
+  public int propertyTryAsInt(String property, int defaultValue) {
+    return NumberUtils.toInt(property, defaultValue);
```

Comment:
Looks like we do not want unnecessary dependencies, and commons.lang3 was explicitly removed once:  https://github.com/apache/iceberg/issues/2100.  so let's just manually code it here

Reply:
Removed and changed to something similar in PropertyUtil, but I do realize that we have a few places use the StringUtil from `apache.commons.lang3` per https://github.com/search?q=repo%3Aapache%2Ficeberg%20import%20org.apache.commons.lang3&type=code,

Reply:
Yea I think it is unfortunate it crept in, but I guess we could remove it later.